tests/unit/dataactvalidator/test_b9_award_financial.py

```diff
@@ -0,0 +1,71 @@
+                               allocation_transfer_agency='test', main_account_code='test',
+                               program_activity_name='test', program_activity_code='test')
+
+    pa = ProgramActivityFactory(budget_year=2016, agency_id='test', allocation_transfer_id='test',
+                                account_number='test', program_activity_name='test', program_activity_code='test')
+
+    assert number_of_errors(_FILE, database, models=[af, pa]) == 0
+
+    # program activity name/code as null
+    af = AwardFinancialFactory(row_number=1, beginning_period_of_availa=2016, agency_identifier='test',
+                               allocation_transfer_agency='test', main_account_code='test',
+                               program_activity_name=None, program_activity_code=None)
+
+    pa = ProgramActivityFactory(budget_year=2016, agency_id='test', allocation_transfer_id='test',
+                                account_number='test')
+
+    assert number_of_errors(_FILE, database, models=[af, pa]) == 0
+
+
+def test_failure(database):
+    """ Testing invalid program activity name for the corresponding TAS/TAFS as defined in Section 82 of OMB Circular
+    A-11. """
+
+    af = AwardFinancialFactory(row_number=1, beginning_period_of_availa=2016, agency_identifier='test',
+                               allocation_transfer_agency='test', main_account_code='test',
+                               program_activity_name='test_wrong', program_activity_code='test')
+
+    pa = ProgramActivityFactory(budget_year=2016, agency_id='test', allocation_transfer_id='test',
+                                account_number='test', program_activity_name='test', program_activity_code='test')
+
+    assert number_of_errors(_FILE, database, models=[af, pa]) == 1
+
+    af = AwardFinancialFactory(row_number=1, beginning_period_of_availa=2016, agency_identifier='test',
+                               allocation_transfer_agency='test', main_account_code='test',
+                               program_activity_name='test', program_activity_code='test_wrong')
+
+    pa = ProgramActivityFactory(budget_year=2016, agency_id='test', allocation_transfer_id='test',
+                                account_number='test', program_activity_name='test', program_activity_code='test')
+
+    assert number_of_errors(_FILE, database, models=[af, pa]) == 1
+
+
+    # program activity name/code as null
+    af = AwardFinancialFactory(row_number=1, beginning_period_of_availa=2016, agency_identifier='test_wrong',
+                               allocation_transfer_agency='test', main_account_code='test',
+                               program_activity_name=None, program_activity_code=None)
+
+    pa = ProgramActivityFactory(budget_year=2016, agency_id='test', allocation_transfer_id='test',
+                                account_number='test')
+
+    assert number_of_errors(_FILE, database, models=[af, pa]) == 1
```

Comment:
The SQL seems to imply that we're comparing each `award_financial` with _other_ `award_financial`s -- should there be a test that includes creating multiple `award_financials` with different `row_number`s?